src/main/java/org/terasology/gooeyDefence/EnemyManager.java

```diff
@@ -85,6 +108,39 @@ public void onPeriodicActionTriggered(PeriodicActionTriggeredEvent event, Entity
         }
     }
 
+    /**
+     * Called when the a path is changed.
+     *
+     * @see OnPathChanged
+     */
+    @ReceiveEvent
+    public void onPathChanged(OnPathChanged event, EntityRef shrineEntity) {
+        for (EntityRef enemy : enemies) {
+            /* Firstly check if the enemy is on an unchanged path */
+            if (enemy.hasComponent(EntrancePathComponent.class)) {
+                if (enemy.getComponent(EntrancePathComponent.class).getEntranceId() != event.getPathId()) {
```

Comment:
in line with my previous comment (or an upcoming comment, depending on the order you read these), it's confusing that the entranceId is compared with the pathId.